lib/parser.js

```diff
@@ -51,20 +30,18 @@ const EVENT_EMIT_RE = /\bfire\s*\(\s*((?:'[^']*')|(?:"[^"]*")|(?:`[^`]*`))/;
 
 class Parser extends EventEmitter {
     constructor(options) {
-        options = Object.assign({}, DEFAULT_OPTIONS, options);
+        super();
 
         Parser.validateOptions(options);
 
-        super();
-
         this.source = options.source;
-        this.features = options.features || SUPPORTED_FEATURES;
+        this.features = options.features;
 
         if (hasOwnProperty(options.source, 'script') && options.source.script) {
             this.scriptOffset = options.source.scriptOffset || 0;
 
             try {
-                this.ast = espree.parse(options.source.script, options);
+                this.ast = espree.parse(options.source.script, options.ast);

```

Comment:
Just to clarify for me. This change is breacking for v2 parser (it requires to mention in changelog and readme files), because now espree options passed from `ast` property instead of passing the whole object. Please correct me if I'm wrong.

Reply:
TL; DR; I don't think it's a breaking change but it could still be mentioned.

It would only be breaking if some of the v2 ast options are suppose to be set by the library user. Is this the case? For example, can you expect this:

```javascript
parser.parse({
  // ...
  comment: false
  ecmaVersion: 6
}
```

to happen? Because from what I can see in the code, you can't generate `sourceCode` without at least `loc`, `range`, and `tokens` set to `true`. And changing the `ecmaVersion` or `ecmaFeatures` could break the parser, so I don't think any of the ast options can be different. There was a `attachComment` options, but it was never used in the parser. That's why I put all the espree options in `options.ast`. Even more so now that there is typings for the main parse function, library users should not set any of the espree options directly.

Reply:
Thanks!

Reply:
I ended up not using `options.ast` and instead call `getAstDefaultOptions` whenever the options are needed.

Reply:
Great